app/models/tag_color_manipulator.rb

```diff
@@ -0,0 +1,15 @@
+class TagColorManipulator
```

Comment:
I'm not well versed in PORO design so let me know whatever you want changed here